### ISSUE TITLE
Standardize basic contribution rules across all repos and basic repository settings

### ISSUE THREAD

**Issue by derberg** [asyncapi/asyncapi#312]
Proposal:

### Forks only

Doesn't matter who you are, org member or not, all work on AsyncAPI should be done in Forks. PRs should come from Forks, and there should be no feature branches in the upstream.

**Why**
- A common practice for larger projects
- Get rid of issue of polluted list of branches, and zombie branches that were already merged.

**What has to be done**
- Clear instruction on how to setup your Fork so it is up to date with upstream master
  - explain the usage of [Pull bot](https://probot.github.io/apps/pull/)
  - explain how to set your local fork in the way that master is pointing to upstream

### Disable Wiki and Projects

**Why**
We do not use this feature, let's just disable it for all repositories.

**What has to be done**
* cleanup if we have some old stuff there
* disable those features and document for future

### Squash merging only

All repos should allow only `squash merging` of PRs

**Why**
To not have `Merge pull request ...` commits that are not very descriptive, but always have a commit message that introduces the commit. The advantage is also that you always have a final word on commit message before merging in.

Example of squash and merge, and just a merge
<img width="718" alt="Screenshot 2020-02-11 at 10 04 00" src="https://user-images.githubusercontent.com/6995927/74222946-e3088780-4cb5-11ea-9a81-367097964a23.png">

**What has to be done**
* set squash and merge  for all repos and document for future

### Introduce CODEOWNERS

It is a GitHub https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners that helps out to automate the assignment of PRs to proper group of people.

**Why**
You can, for example, specify that specific part of the repository can be only reviewed by a given person. If someone donates a template into the generator, he can become a main code owner to assure that only he/she can approve changes in it. 
Code owners is also well integrated with branch protection, so you can set a clear PR requirement

<img width="699" alt="Screenshot 2020-02-11 at 10 13 12" src="https://user-images.githubusercontent.com/6995927/74223527-30392900-4cb7-11ea-8a30-cab32ff0e251.png">

**What has to be done**
* create those files in all repos, specify code owners and then set proper branch protection
* document for future repos

Reason: 
To encourage contributions and tools adoption, we need to make sure work done on those projects is done consistently and in clean way.

**Comment by jonaslagoni** [comment]
> ...
> ### Disable Wiki and Projects
> 
> **Why**
> We do not use this feature, let's just disable it for all repositories.
> ...


Just to mention, we still have an outstanding [issue](https://github.com/asyncapi/generator/issues/196) on the generator in regards to the wiki.

Everything else I agree with :smile:

**Comment by derberg** [comment]
@jonaslagoni Thanks a lot for fast response!!!
Regarding https://github.com/asyncapi/generator/issues/196 - does it really have to be wiki? why not markdown that we would have in new `docs` directory? much easier to collaborate, wiki is not really nice

**Comment by jonaslagoni** [comment]
Yea I have no preferences, so it could might as well be the in a `docs` directory as markdown :+1:

**Comment by derberg** [comment]
Awesome @jonaslagoni !
Will you adjust the issue from the generator repository?

**Comment by derberg** [comment]
Wiki/Projects and `squash&merge` thingy done in all the repos + PR with docs update https://github.com/asyncapi/asyncapi/pull/317

**Comment by derberg** [comment]
Ok, another point addressed - Forking.

I created this PR https://github.com/asyncapi/asyncapi/pull/318 - you might wanna have a look.

---

Next steps after merge:
- I will go through all repos and immediately remove all the branches that are already merged
- I will send an email to all current members of the organization that have `write` access to repositories, to let them know about the change, so they know they have to follow
- for the branches that are in the upstream not merged and:
 - there is no PR - I contact owner about the status
 - there is a PR opened - I leave it as it is

@jonaslagoni @fmvilas agreed? or you think we can do it more efficiently?

**Comment by jonaslagoni** [comment]
Can't see another option, good job :+1:

**Comment by fnobilia** [comment]
@derberg regarding `Get rid of issue of polluted list of branches, and zombie branches that were already merged.`, for the future you could set *automatic deletion of merged branches* on all repos.

Github documentation for reference https://help.github.com/en/github/administering-a-repository/managing-the-automatic-deletion-of-branches

**Comment by derberg** [comment]
@fnobilia you're right, thanks a lot for suggestion, I'll make sure we have it in all repos after merging #318

**Comment by RobertDiebels** [comment]
The above looks excellent to me! I do have one question. Will we be using release-branches a la GitFlow? Or will we continue working with the master branch alone and tag it?

**Comment by derberg** [comment]
@RobertDiebels we are before a major release, so I would not introduce release branches yet. To be more precise, GitFlow is more about master-develop than master-release. We have lightweight GitHubFlow. Master is always stable and works like a charm and whatever is merged should never break the library. 

At this stage of the project,  tagging is enough, until we have major releases and we have to consider something like a bug fix support for older versions.

**Comment by derberg** [comment]
All repos now have setting that branches after merge are automatically deleted 
Anyone wants to have a look and approve? https://github.com/asyncapi/asyncapi/pull/325

**Comment by derberg** [comment]
OK, I consider fork/branch cleanup completed
What left from this one is CODEOWNERS setup

**Comment by derberg** [comment]
Codeowners introduction waits for asyncapi joining the foundation and setting open governance
